### PR TITLE
Introduce tests with jest

### LINKED ISSUE #235: Add Tests to the bot
Currently, there are no automatic tests.

To make sure that working code continues to work, tests should be added.

- [ ] add test framework (jest)
- [ ] write test for https://github.com/Yes-Theory-Fam/yesbot-ts/pull/242 after it is merged.

This will be the foundation, so all further refactoring can write test directly

### PR DESCRIPTION
**Pull request by adrian-goe**
Introduce jest tests with 2 first tests.
Add Testing in pipline

closes #235

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
-/build
+**./*.log
 .env
-/.vscode
+.npmrc
 /.idea
+/.vscode
+/build
 /coverage
+/coverage/
```
Comment (geisterfurz007) [inline .gitignore:8]: Duplicate intended?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -69,15 +69,15 @@ The bot's main code is contained in the `src` directory which contains several s
...
+- programs - Code containing logic for all the commands and features of the bot. This directory contains an `events.ts`
   which reexports all exports from all files in this folder, please do so to, when adding your own feature.
 - scripts - Standalone scripts that are designed to be manually run for one-off tasks (like importing birthdays for
   example)
 
 It also contains two single files:
 
 - prisma.ts - Simple setup of the TypeORM database connection
-- index.ts - Creates and exports the discord.js Client used in the entire application. This is the entry point for the
+- events.ts - Creates and exports the discord.js Client used in the entire application. This is the entry point for the
```
Comment (geisterfurz007) [inline CONTRIBUTING.md:80]: This looks like refactoring relics and probably wasn't intended?

File: src/event-distribution/events/events.ts [hunk 3]
```diff
@@ -27,9 +28,22 @@ export type HandlerFunction<T extends DiscordEvent> =
...
+        >;
+        prototype: any;
+      }
+    | CommandHandler<
+        (MessageEventHandlerOptions | ReactionEventHandlerOptions)["event"]
+      >,
+  tree: StringIndexedHIOCTree<
+    | DiscordEvent.MESSAGE
+    | DiscordEvent.REACTION_ADD
+    | DiscordEvent.REACTION_REMOVE
```
Comment (geisterfurz007) [inline src/event-distribution/events/events.ts:45]: Why explicitly list these again? The argument types are already given through the type of `addEventHandler`.

Comment (adrian-goe) [reply src/event-distribution/events/events.ts:45]: i Iwanted t copy this explicit types for the mock and forgot to revert 🙈

**Review by geisterfurz007 — APPROVED** [review]
LGTM!

**Pull request merged**